cloudamqp/resource_cloudamqp_rabbitmq_configuration.go

```diff
@@ -0,0 +1,240 @@
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(int)
+					if v < 10000 || v > 25000000 {
+						errs = append(errs, fmt.Errorf("%q must be between 10000 and 25000000 inclusive, got: %d", key, v))
+					}
+					return
+				},
+			},
+			"vm_memory_high_watermark": {
+				Type:        schema.TypeFloat,
+				Optional:    true,
+				Default:     0.81,
+				Description: "When the server will enter memory based flow-control as relative to the maximum available memory.",
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(float64)
+					if v < 0.4 || v > 0.9 {
+						errs = append(errs, fmt.Errorf("%q must be between 0.4 and 0.9 inclusive, got: %d", key, v))
+					}
+					return
+				},
+			},
+			"queue_index_embed_msgs_below": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     4096,
+				Description: "Size in bytes below which to embed messages in the queue index.",
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(int)
+					if v < 1 || v > 10485760 {
+						errs = append(errs, fmt.Errorf("%q must be between 1 and 10485760 inclusive, got: %d", key, v))
+					}
+					return
+				},
+			},
+			"max_message_size": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     134217728,
+				Description: "The largest allowed message payload size in bytes.",
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(int)
+					if v < 1 || v > 536870912 {
+						errs = append(errs, fmt.Errorf("%q must be between 1 and 536870912 inclusive, got: %d", key, v))
+					}
+					return
+				},
+			},
+			"log_exchange_level": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "error",
+				Description: "Log level for the logger used for log integrations and the CloudAMQP Console log view. " +
+					"Does not affect the file logger. Requires a RabbitMQ restart to be applied.",
+				ValidateFunc: validateLogLevel(),
+			},
+		},
+	}
+}
+
+func resourceRabbitMqConfigurationCreate(d *schema.ResourceData, meta interface{}) error {
+	id := strconv.Itoa(d.Get("instance_id").(int))
+	d.SetId(id)
+	return resourceRabbitMqConfigurationUpdate(d, meta)
+}
+
+func resourceRabbitMqConfigurationRead(d *schema.ResourceData, meta interface{}) error {
+	api := meta.(*api.API)
+	instanceID, _ := strconv.Atoi(d.Id())
+	data, err := api.ReadRabbitMqConfiguration(instanceID)
+	log.Printf("[DEBUG] cloudamqp::resource::rabbit_configuration::read data: %v", data)
+	if err != nil {
+		return err
+	}
+
+	d.Set("instance_id", instanceID)
+	for k, v := range data {
+		if validateRabbitMqConfigurationJSONField(k) {
+			key := strings.ReplaceAll(k, "rabbit.", "")
+			if key == "connection_max" {
+				if v == "infinity" || v == nil {
+					v = -1
```

Comment:
Just recording the motivation behind this: so we can use `schema.TypeInt` for this property